src/App.tsx

```diff
@@ -15,36 +15,91 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE = '',
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+}
+
+function getPreparedGoods(goods: string[], sortParameter: SortType): string[] {
+  if (sortParameter) {
+    goods.sort((a, b) => {
+      switch (sortParameter) {
+        case SortType.ALPHABET:
+          return a.localeCompare(b);
+        case SortType.LENGTH:
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return goods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortParameter, setSortParameter] = useState(SortType.NONE);
+
+  const goods = getPreparedGoods([...goodsFromServer], sortParameter);
+
+  if (isReversed) {
+    goods.reverse();
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => {
+            setSortParameter(SortType.ALPHABET);
+          }}
+          type="button"
+          className={`button is-info ${sortParameter !== SortType.ALPHABET ? 'is-light' : ''}`}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => {
+            setSortParameter(SortType.LENGTH);
+          }}
+          type="button"
+          className={`button is-success ${sortParameter !== SortType.LENGTH ? 'is-light' : ''}`}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={() => {
+            setIsReversed(!isReversed);
```

Comment:
For dynamic state changes - use functional setState
```suggestion
            setIsReversed((prevState) => !prevState);
```